Cluster labelling wiring

ubergesundheit opened the pull request:
Followup to #1474 and #1499  Implements giantswarm/giantswarm#10541

This also adds some tests for cluster labels.

I tried to follow the new approach introduced in #1462

ubergesundheit commented on src/reducers/index.js:
```diff
@@ -18,13 +19,14 @@ import users from './userReducer';
   catalogs,
+  clusterLabels,
   clusters,
   credentials,
   invitations,
+  nodePools,
   organizations,
   releases,
   users,
-  nodePools,
```
Sorted a-z. Or this the order intentional?

axbarsan replied:
Let's hope not 😬

axbarsan commented on src/components/Cluster/ClusterDetail/ClusterLabels/ClusterLabels.tsx:
```diff
@@ -2,19 +2,25 @@ import styled from '@emotion/styled';
 import EditLabelTooltip from './EditLabelTooltip';
 
-interface IClusterLabelsProps {
+interface IClusterLabelsProps extends HTMLAttributes<HTMLDivElement> {
```
`HTMLAttributes` is not exactly the most suitable type because those are HTML attributes, not React attributes (some attributes have different names)
```suggestion
interface IClusterLabelsProps extends React.ComponentPropsWithoutRef<'div'> {
```

axbarsan commented on src/stores/clusterlabels/reducer.ts:
```diff
@@ -0,0 +1,36 @@
+      draft.requestInProgress = true;
+      draft.error = null;
+
+      return;
```
```suggestion
      break;
```

ubergesundheit replied:
`return` vs `break` here: Does this change anything since the other cases don't get executed either way?

axbarsan replied:
It does not change the behaviour of the code. It's just a matter of using the correct syntax for the `switch` statement
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

axbarsan commented on src/stores/clusterlabels/reducer.ts:
```diff
@@ -0,0 +1,36 @@
+      draft.requestInProgress = false;
+      draft.error = null;
+
+      return;
```
```suggestion
      break;
```

axbarsan commented on __tests__/V5ClusterManagement.js:
```diff
@@ -447,3 +449,127 @@ it('renders an error message if there was an error loading apps', () => {
+  const { findByText, findByTestId } = renderRouteWithStore(clusterDetailPath);
+
+  await findByText('Labels:');
+  fireEvent.click(await findByTestId('add-label-button'));
+
+  const keyInput = await findByTestId('label-key-input');
+  const valueInput = await findByTestId('label-value-input');
+  const saveButton = await findByTestId('save-label-button');
```
What do you think about using more accessible queries? We had a discussion about this at some point, I believe before you joined:

These are the rules we usually guide by:
https://testing-library.com/docs/guide-which-query

Also, I see that you're using a lot of async queries. Are all of those really necessary, or can we replace them with sync ones? They have a big impact on the duration of the tests

ubergesundheit commented:
@axbarsan care to review again? I've taken your advice and changed the tests to use less promise based selectors & also changed all selectors to be by text, role or labeltext
